File: Python3/1 - 200/119 Pascal's Triangle II.py
# 32ms, 100.0%
class Solution:
    def getRow(self, rowIndex):
        res = []
        for i in range(-1, rowIndex):
            res.append(1)
            while i > 0:
                res[i] = res[i - 1] + res[i]
                i -= 1
        return res
    

File: 1 - 200/119. Pascal's Triangle II.cc
// 3ms, 0.55%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> ans(rowIndex + 1, 1);
        int num = 0;
        while (++num < rowIndex)
            for (int i = num; i > 0; --i) ans[i] += ans[i - 1];
        return ans;
    }
};
